### ISSUE TITLE
Identify steam EOF

### ISSUE THREAD

**Issue by kerenbCyber** [sshnet/SSH.NET#758]
Hi,

We are using Renci for SSH communication, 
Using "string Read()" function for reading the prompt. 

How can we check if we receive EOF of the steam (whether or not the spawned process has ended.)?
 
Some devices, when changing their passwords or performing a commit command would close the connection and force you to reconnect - this is the prompt we would use to recognize the event (EOF)

Thanks,
Keren

**Comment by IgorMilavec** [comment]
Can you provide a code sample?

**Comment by IgorMilavec** [comment]
Closed due to inactivity

**Comment by glorialearnscoding** [comment]
Hello @IgorMilavec , I have a similar question when using ShellStream.Expect(). When using the Cisco device I have to use the ShellStream to write our own command execution and result matching functions. I'm using a match pattern of
string PromptPattern = @"([\w-]+\s*(\([\w-]+\))?\s*\#)";
to get the expected result from ShellStream: expecting a successful result should be:
```
<command>
<result>
prompt#
```
However, sometimes I got a result with a # in the middle, and then my match pattern thinks # as a prompt and ended the expect().

So it would be great if I can find something like EOF to indicate that the execution is finished (either succeess or fail). I did a search and found the SSH_MSG_CHANNEL_EOF message. However, this looks like something used internally. Is there some method to use it as a customer? Thanks!
